diff --git a/code/builders/builder-vite/src/vite-final.test.ts b/code/builders/builder-vite/src/vite-final.test.ts
index b1dc2bf7fcb..6940cc9c86c 100644
--- a/code/builders/builder-vite/src/vite-final.test.ts
+++ b/code/builders/builder-vite/src/vite-final.test.ts
@@ -1,7 +1,5 @@
 import { afterEach, beforeEach, describe, expect, it, vi } from 'vitest';
 
-import type { Options } from 'storybook/internal/types';
-
 import type { InlineConfig } from 'vite';
 
 vi.mock('vite', async (importOriginal) => {
@@ -28,27 +26,7 @@ vi.mock('vite', async (importOriginal) => {
 const { mergeConfig } = await import('vite');
 const mergeConfigMock = vi.mocked(mergeConfig);
 
-async function viteFinal(
-  config: InlineConfig,
-  options: Pick<Options, 'configType'>
-): Promise<InlineConfig> {
-  const extraConfig: InlineConfig = {};
-
-  if (options.configType === 'DEVELOPMENT') {
-    extraConfig.resolve = {
-      alias: {
-        'storybook/theming': '/virtual/storybook/theming',
-        'storybook/theming/create': '/virtual/storybook/theming/create',
-      },
-    };
-  }
-
-  extraConfig.build = {
-    sourcemap: process.env.CI ? false : true,
-  };
-
-  return mergeConfig(config, extraConfig);
-}
+const { viteFinal } = await import('./vite-final.ts');
 
 describe('viteFinal', () => {
   const baseConfig: InlineConfig = {
diff --git a/code/builders/builder-vite/src/vite-final.ts b/code/builders/builder-vite/src/vite-final.ts
new file mode 100644
index 00000000000..19c1fd1bc23
--- /dev/null
+++ b/code/builders/builder-vite/src/vite-final.ts
@@ -0,0 +1,27 @@
+import type { Options } from 'storybook/internal/types';
+
+import type { InlineConfig } from 'vite';
+
+import { mergeConfig } from 'vite';
+
+export async function viteFinal(
+  config: InlineConfig,
+  options: Pick<Options, 'configType'>
+): Promise<InlineConfig> {
+  const extraConfig: InlineConfig = {};
+
+  if (options.configType === 'DEVELOPMENT') {
+    extraConfig.resolve = {
+      alias: {
+        'storybook/theming': '/virtual/storybook/theming',
+        'storybook/theming/create': '/virtual/storybook/theming/create',
+      },
+    };
+  }
+
+  extraConfig.build = {
+    sourcemap: process.env.CI ? false : true,
+  };
+
+  return mergeConfig(config, extraConfig);
+}
\ No newline at end of file
